aea/connections/base.py

```diff
@@ -46,9 +46,9 @@ class Connection(ABC):
 
     def __init__(
         self,
-        connection_id: str,
-        restricted_to_protocols: Optional[Set[str]] = None,
-        excluded_protocols: Optional[Set[str]] = None,
+        connection_id: Optional[PublicId] = None,
```

Comment:
This is one of the big changes in the PR.

`class Connection` now requires a `PublicId`. If not specified by the subclass constructor, the `Connection.__init__` raises an error (see the `ValueError` below).
Also, removed every default connection_id in subclasses of `Connection`. This should help in using the same code in other packages without touching the modules. Also, this only regards the programmatic usage of a connection, whereas if only the CLI is used this change is completely hidden to the user.

Reply:
analogous choices made for `restricted_to_protocols` and `excluded_protocols`. 
Indeed, if the user isnot interested in the programmatic usage, they should only be customized at configuration level, and not a code level.